# Generated by Django 4.2 on 2025-03-18 06:21

from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Genome',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('bin_id', models.CharField(db_index=True, help_text='Unique bin identifier, e.g., SRR28393034_bin.2', max_length=100, unique=True)),
                ('source', models.CharField(db_index=True, help_text='Source identifier, e.g., SRR28393034', max_length=100)),
                ('habitat', models.CharField(blank=True, db_index=True, help_text='Habitat type, e.g., Mangrove', max_length=200, null=True)),
                ('marker_lineage', models.CharField(blank=True, help_text='Marker lineage used for quality assessment, e.g., p__Proteobacteria (UID3880)', max_length=200, null=True)),
                ('completeness', models.FloatField(blank=True, help_text='Genome completeness percentage (0-100)', null=True)),
                ('contamination', models.FloatField(blank=True, help_text='Genome contamination percentage (0-100)', null=True)),
                ('strain_heterogeneity', models.FloatField(blank=True, help_text='Strain heterogeneity percentage (0-100)', null=True)),
                ('gtdb_classification', models.TextField(blank=True, help_text='GTDB taxonomy classification', null=True)),
                ('closest_genome_reference', models.CharField(blank=True, help_text='Closest reference genome identifier', max_length=200, null=True)),
                ('closest_genome_ani', models.FloatField(blank=True, help_text='Average Nucleotide Identity to closest reference genome', null=True)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
            ],
            options={
                'ordering': ['source', 'bin_id'],
            },
        ),
        migrations.AddIndex(
            model_name='genome',
            index=models.Index(fields=['bin_id'], name='genome_geno_bin_id_646dc4_idx'),
        ),
        migrations.AddIndex(
            model_name='genome',
            index=models.Index(fields=['source'], name='genome_geno_source_cc1c9b_idx'),
        ),
        migrations.AddIndex(
            model_name='genome',
            index=models.Index(fields=['habitat'], name='genome_geno_habitat_afbda1_idx'),
        ),
        migrations.AddIndex(
            model_name='genome',
            index=models.Index(fields=['completeness'], name='genome_geno_complet_da4cc8_idx'),
        ),
        migrations.AddIndex(
            model_name='genome',
            index=models.Index(fields=['contamination'], name='genome_geno_contami_a1bb05_idx'),
        ),
    ]
